zhaquirks/tuya/ts0601_gas.py

```diff
@@ -1,123 +1,55 @@
-                )
-            else:
-                self.endpoint.device.ias_bus.listener_event("update_zone_status", 0)
-        else:
-            _LOGGER.warning(
-                "[0x%04x:%s:0x%04x] unhandled attribute: 0x%04x",
-                self.endpoint.device.nwk,
-                self.endpoint.endpoint_id,
-                self.cluster_id,
-                attrid,
-            )
-
-
-class TuyaGasDetectorZone(CustomCluster, IasZone):
-    """IAS Zone."""
-
-    _CONSTANT_ATTRIBUTES = {ZONE_TYPE: IasZone.ZoneType.Carbon_Monoxide_Sensor}
-
-    def __init__(self, *args, **kwargs):
-        """Init."""
-        super().__init__(*args, **kwargs)
-        self.endpoint.device.ias_bus.add_listener(self)
-
-    def update_zone_status(self, value):
-        """Update IAS status."""
-        super()._update_attribute(ZONE_STATUS, value)
-
-
-class TuyaGasDetector0601(CustomDevice):
-    """TS0601 _TZE200_ggev5fsl quirk."""
-
-    def __init__(self, *args, **kwargs):
-        """Init."""
-        self.ias_bus = Bus()
-        super().__init__(*args, **kwargs)
-
-    signature = {
-        MODELS_INFO: [
-            ("_TZE200_ggev5fsl", "TS0601"),
-            ("_TZE200_rjxqso4a", "TS0601"),
-        ],
-        ENDPOINTS: {
-            1: {
-                PROFILE_ID: zigpy.profiles.zha.PROFILE_ID,
-                DEVICE_TYPE: zigpy.profiles.zha.DeviceType.SMART_PLUG,
-                INPUT_CLUSTERS: [
-                    Basic.cluster_id,
-                    Groups.cluster_id,
-                    Scenes.cluster_id,
-                    TuyaManufCluster.cluster_id,
-                ],
-                OUTPUT_CLUSTERS: [
-                    Time.cluster_id,
-                    Ota.cluster_id,
-                ],
-            },
-        },
-    }
-
-    replacement = {
-        ENDPOINTS: {
-            1: {
-                PROFILE_ID: zigpy.profiles.zha.PROFILE_ID,
-                DEVICE_TYPE: zigpy.profiles.zha.DeviceType.IAS_ZONE,
-                INPUT_CLUSTERS: [
-                    Basic.cluster_id,
-                    Groups.cluster_id,
-                    Scenes.cluster_id,
-                    TuyaGasDetectorZone,
-                    TuyaGasDetectorCluster,
-                ],
-                OUTPUT_CLUSTERS: [
-                    Time.cluster_id,
-                    Ota.cluster_id,
-                ],
-            },
-        },
-    }
+    .tuya_enum(
+        dp_id=9,
+        attribute_name="self_test_result",
+        enum_class=TuyaSelfTestResult,
+        entity_type=EntityType.DIAGNOSTIC,
+        entity_platform=EntityPlatform.SENSOR,
+        translation_key="self_test_result",
+        fallback_name="Self test result",
+    )
+    .tuya_battery(dp_id=15, power_cfg=TuyaPowerConfigurationCluster2AA)
+    .tuya_switch(
+        dp_id=16,
+        attribute_name="silence",
+        entity_type=EntityType.STANDARD,
+        translation_key="silence",
+        fallback_name="Silence",
+    )
```

Comment:
I'm wondering if "Mute" or "Mute siren" would be better for this? Not sure if we already have similar entities for this.